src/utilities/strategy-helpers/deployment-helper.ts

```diff
@@ -31,19 +35,51 @@ export async function deploy(kubectl: Kubectl, manifestFilePaths: string[], depl
     inputManifestFiles = updateImagePullSecretsInManifestFiles(inputManifestFiles, TaskInputParameters.imagePullSecrets);
 
     // deployment
-    const deployedManifestFiles = deployManifests(inputManifestFiles, kubectl, isCanaryDeploymentStrategy(deploymentStrategy));
+    const deployedManifestFiles = deployManifests(inputManifestFiles, kubectl, isCanaryDeploymentStrategy(deploymentStrategy), isBlueGreenDeploymentStrategy(deploymentStrategy));
 
     // check manifest stability
     const resourceTypes: Resource[] = KubernetesObjectUtility.getResources(deployedManifestFiles, models.deploymentTypes.concat([KubernetesConstants.DiscoveryAndLoadBalancerResource.service]));
     await checkManifestStability(kubectl, resourceTypes);
 
+    // route blue-green deployments
+    if (blueGreenDeploymentHelper.isBlueGreenDeploymentStrategy()) {
+        await routeBlueGreen(kubectl, inputManifestFiles);
+    }
+
     // print ingress resources
     const ingressResources: Resource[] = KubernetesObjectUtility.getResources(deployedManifestFiles, [KubernetesConstants.DiscoveryAndLoadBalancerResource.ingress]);
     ingressResources.forEach(ingressResource => {
         kubectl.getResource(KubernetesConstants.DiscoveryAndLoadBalancerResource.ingress, ingressResource.name);
     });
 }
 
+async function routeBlueGreen(kubectl: Kubectl, inputManifestFiles: string[]) {
+    // get buffer time
+    let sleepTime: number = TaskInputParameters.versionSwitchBuffer;
+    // check if proper input
+    if(sleepTime<0 || sleepTime>300) {
+        console.log('Max buffer time is 300 minutes');
+        throw('TooLongBufferTime')
+    }
+
+    let temp = new Date();
+    console.log('starting buffer time of '+sleepTime+' minute/s at '+temp.getHours()+':'+temp.getMinutes()+':'+temp.getSeconds()+' UTC');
+    // waiting

```

Comment:
Move the console log statements to `sleep` function.
Where you log the start time and end time

Reply:
im using an already existing sleep function from utils which is used in manifest stability file where i dont think logging is a good idea. should i create another function?